### PR TITLE
Implement constant-time secret comparisons in CEL.

### PR DESCRIPTION
**Pull request by bigkevmcd**
# Changes

This adds new compareSecret overload for strings in CEL, allowing comparison of a value against a secret.

This addresses #486 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```
New CEL interceptor function `compareSecrets` for securely comparing strings to secrets in CEL expressions.
```

### CODE REVIEW

File: docs/cel_expressions.md [hunk 1]
```diff
@@ -206,4 +206,19 @@ interceptor.
...
+    </th>
+    <td>
+      string.compareSecret(string, string, string) -> bool
+    </td>
+    <td>
+      Constant-time comparison of strings against secrets, this will fetch the secret using the combination of namespace/name and compare the token key to the string using a cryptographic constant-time comparison..<p>
+      The event-listener service account must have access to the secret.
+    </td>
+    <td>
+     <pre>header.canonical('X-Secret-Token').compareSecret('namespace', 'secret-name', 'key')</pre>
```
Comment (dibyom) [inline docs/cel_expressions.md]: can the namespace be optional? In other interceptors, if one is not specified, it defaults to the EL's namespace.

Comment (bigkevmcd) [reply docs/cel_expressions.md]: I implemented it with variable args, if you provide 3, the first is the namespace, only 2 and it defaults to the eventlistener's namespace.

Comment (dibyom) [reply docs/cel_expressions.md]: Ahh ok, that sounds good.

**Comment by dibyom** [comment]
/approve

One minor thing about the optional namespace position in the function signature but other wise 👍

**Pull request merged**